### PR TITLE
Switch from pelican to doxygen.

### PR DESCRIPTION
**Pull request by zoq**
Attemp to move from pelican to doxygen, which enables us to use the same design for the blog as for the main site.

Here is a demo: http://kurg.org/misc/blog/index.html

### CODE REVIEW

**Comment by rcurtin** [comment]
Looks great to me.  I can see that there might be a little bit of CSS work to handle but that should not be hard.  The only big issue I see so far is that the author is down at the bottom, but I feel like it should be at the top.  We can try to use CSS to fix this or we can try and modify the file so that the author is printed below the title---what do you think would be easier?  I'm happy to give either a shot.

It would also be cool to get the author in the page list:
http://kurg.org/misc/blog/pages.html
but I think maybe the easiest way to do that is with some postprocessing script (maybe the same is true for getting the author name on the page).

Let me know what you think.

**Comment by zoq** [comment]
Since Doxygen isn't meant to be used as a blog engine, some python code generates ``authors.html, index.html, and pages.html``

- adjust-author.py
- build-authors.py
- build-archive.py
- build-index.py

Also, there is another issue; Doxygen ignores the ``@author`` tag. To solve the issue I preprocess each post (transform the author tag into a temporary snippet which is removed later). Anyway, let me know what you think about the updated author information. I'm not fully satisfied with ``pages.html``.

**Review by rcurtin — COMMENTED** [review]
I think the author changes are really nice.  Maybe a little CSS work could clean the pages.html page up, I'm not sure.  If you like, we can merge this and then I can take a shot at the CSS.  I think I already said this, but I think overall it looks really nice, a big improvement. :)

File: script/build-documentation.sh [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
-doxygensrcdir=../script/doxygen-src
+# git pull, then rebuild the docs.
+#
+# $1: version of mlpack (should be unpacked into /var/www/www.mlpack.org/mlpack-x.y.z/)
+# should be in the format "mlpack-2.2.3" or similar.  "mlpack-git" for git
+# master.
+srcdir=/Volumes/koch/src/blog/
+docdir=/Volumes/koch/src/blog/script/$1/
+postdir=/Volumes/koch/src/blog/script/doxygen-post/
+doxygensrcdir=/Volumes/koch/src/blog/script/doxygen-src
```
[the remark this answers is not in the document]
Comment (zoq) [reply script/build-documentation.sh]: Absolutely, accidentally pushed the changes here: https://github.com/mlpack/blog/pull/2/commits/bc3a7011ac63aa94da3bfe01d5fce5be7bfd234d

**Comment by zoq** [comment]
Once this is merged, I'll go and adjust the jenkins job configuration.

**Pull request merged**